src/Playwright/Core/BrowserType.cs

```diff
@@ -117,5 +121,36 @@ public async Task<IBrowserContext> LaunchPersistentContextAsync(string userDataD
                 recordHarOmitContent: options.RecordHarOmitContent,
                 recordVideo: Browser.GetVideoArgs(options.RecordVideoDir, options.RecordVideoSize)).ConfigureAwait(false)).Object;
         }
+
+        public async Task<IBrowser> ConnectAsync(string wsEndpoint, BrowserTypeConnectOptions options = null)
+        {
+            options ??= new BrowserTypeConnectOptions();
+
+            var loggerFactory = LoggerFactory.Create(builder =>
+            {
+                builder.SetMinimumLevel(LogLevel.Debug);
+                builder.AddFilter((f, _) => f == "PlaywrightSharp.Playwright");
+            });
+            var webSocketTransport = new WebSocketTransport(wsEndpoint, options, loggerFactory);
+            await webSocketTransport.ConnectAsync().ConfigureAwait(false);
```

Comment:
Since (I assume) we always have to call `ConnectAsync`, something like a `static WebSocketTransport.Connect : WebsocketTransport` would feel more natural.

Reply:
yes not required though since when a WebSocket object is created it will create connection & give back instance of websocket